cf-promises/cf-promises.c

```diff
@@ -132,10 +132,13 @@ int main(int argc, char *argv[])
     if (format == GENERIC_AGENT_CONFIG_COMMON_POLICY_OUTPUT_FORMAT_CF ||
         format == GENERIC_AGENT_CONFIG_COMMON_POLICY_OUTPUT_FORMAT_JSON)
     {
+        // If no file was provided, use 'promises.cf' by default
+        if (config->input_file == NULL) {
+            GenericAgentConfigSetInputFile(config, GetInputDir(), "promises.cf");
+        }
+
         // Just parse and write content to output
-        Policy *output_policy = ParserParseFile(AGENT_TYPE_COMMON, config->input_file,
-                                                config->agent_specific.common.parser_warnings,
-                                                config->agent_specific.common.parser_warnings_error);
+        Policy *output_policy = Cf3ParseFile(config, config->input_file);
```

Comment:
Since https://github.com/cfengine/core/commit/2a7c11b4789f876bc36e42d8b473936f56757162, the single file parsing is done before parsing the whole policy, using ParserParseFile directly (which does not handle empty file name and segfaults).

The previous behavior was to always call `LoadPolicy` (which checks if the file exists, and uses "promises.cf" if no file is given) before calling `ParserParseFile`. Another issue with calling directly `ParserParseFile` is that it is not able to read json policy files, contrary to the Higher-level `Cf3ParseFile` which is basically a wrapper around `ParserParseFile` that handles checks and detection of the target file.